azurerm/resource_arm_hd_insight_clusters.go

```diff
@@ -0,0 +1,918 @@
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"domain": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"domain_user_password": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"domain_username": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"ldaps_urls": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeList,
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+						},
+						"organizational_unit_dn": {
+							Optional: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+					},
+				},
+			},
+			"storage_account": {
+				Required: true,
+				ForceNew: true,
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"blob_endpoint": {
+							Required: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"container": {
+							Required: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+						"access_key": {
+							Required: true,
+							ForceNew: true,
+							Type:     schema.TypeString,
+						},
+					},
+				},
+			},
+			"tags": tagsSchema(),
+			"tier": {
+				Optional: true,
+				ForceNew: true,
+				Type:     schema.TypeString,
+				Default:  "standard",
+				ValidateFunc: validation.StringInSlice([]string{
+					"standard",
+					"premium",
+				}, true),
+				DiffSuppressFunc: ignoreCaseDiffSuppressFunc,
+			},
+		},
+	}
+}
+
+func resourceArmHDInsightClustersCreate(d *schema.ResourceData, meta interface{}) error {
+	client := meta.(*ArmClient).hdInsightClustersClient
+	ctx := meta.(*ArmClient).StopContext
+
+	resourceGroupName := d.Get("resource_group_name").(string)
+	clusterName := d.Get("name").(string)
+	parameters := hdinsight.ClusterCreateParametersExtended{}
+	parameters.Location = utils.String(azureRMNormalizeLocation(d.Get("location").(string)))
+	tags := d.Get("tags").(map[string]interface{})
+	tmpParamOfTags := expandTags(tags)
+	parameters.Tags = tmpParamOfTags
+	parameters.Properties = &hdinsight.ClusterCreateProperties{}
+	if paramValue, paramExists := d.GetOk("cluster_version"); paramExists {
+		parameters.Properties.ClusterVersion = utils.String(paramValue.(string))
+	}
+	if paramValue, paramExists := d.GetOk("os_type"); paramExists {
+		parameters.Properties.OsType = hdinsight.OSType(paramValue.(string))
+	}
+	if paramValue, paramExists := d.GetOk("tier"); paramExists {
+		parameters.Properties.Tier = hdinsight.Tier(paramValue.(string))
+	}
+	parameters.Properties.ClusterDefinition = &hdinsight.ClusterDefinition{}
+	if paramValue, paramExists := d.GetOk("cluster_type"); paramExists {
+		parameters.Properties.ClusterDefinition.Kind = utils.String(paramValue.(string))
+	}
```

Comment:
can we update this to match the convention:

```
# ..
name := d.Get("name").(string)
osType := d.Get("os_type").(string)
parameters := hdinsight.ClusterCreateParametersExtended{
  Properties:  &hdinsight.ClusterCreateProperties{
    OsType: hdinsight.OSType(osType.(string)),
    # ..
  },
  Tags: expandTags(tags),
}
```

Reply:
This is base on an auto-generation tool which handles all the marshalling and unmarshalling stuffs.